app/controllers/contributions_controller.rb

```diff
@@ -1,10 +1,34 @@
 class ContributionsController < ApplicationController
   def create
+  	@project = Project.find(params[:project_id])
+  	@contribution = Contribution.new
+  	authorize @contribution
+  	@contribution.project = @project
+  	if @contribution.save
+  	  redirect_to project_path(@contribution)
+  	else
+  	  render 'contributions/edit'
```

Comment:
Creating/editing contributions will take place on the projects#show page, not on a separate contributions/edit page? So the projects#show is what we should render. Same for line 21